packages/api-gateway/src/index.js

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable no-console */
+// import { preflight } from '@web3-storage/worker-utils/cors'
+import { notFound } from '@web3-storage/worker-utils/response'
+import { Router } from '@web3-storage/worker-utils/router'
+import * as dagCbor from '@ipld/dag-cbor'
+import * as dagUcan from '@ipld/dag-ucan'
+import * as carTransport from '@ucanto/transport/car'
+import toBuffer from 'it-to-buffer'
+import toIt from 'browser-readablestream-to-it'
+// @ts-ignore
+// eslint-disable-next-line no-unused-vars
+import * as ucanto from '@ucanto/interface'
+// import { version } from './routes/version.js'
+
+/**
+ * @typedef {import('./bindings.js').ModuleWorker} ModuleWorker
+ */
+
+/**
+ * @param {Request} request
+ * @returns {import('./bindings.js').RouteContext}
+ */
+export function routeContextFromRequest(request) {
+  return {
+    url: new URL(request.url),
+  }
+}
+
+/**
+ * UCAN from http request authorization bearer token
+ */
+// @ts-ignore
+// eslint-disable-next-line no-unused-vars
+class UcanBearerToken {
+  /**
+   * @param {string} header - http authorization header value
+   * @returns {UcanBearerToken}
+   */
+  static fromHeader(header) {
+    return new UcanBearerToken(header.replace(/^Bearer /, ''))
+  }
+
+  /**
+   * @param {string} token
+   * @protected
+   */
+  constructor(token) {
+    if (!token) {
+      throw new Error(`cannot create UcanBearerToken from empty token`)
+    }
+    this.token = token
+    this.parsed = dagUcan.decode(Buffer.from(token))
+  }
+}
+
+/**
+ * @type {Record<string, (...args: any[]) => Promise<ucanto.Invocation>>}
+ */
+const UcantoInvocation = {
+  /**
+   * @param {Request} request
+   * @returns {Promise<ucanto.Invocation>}
+   */
+  fromRequest: async (request) => {
+    const invocations = await carTransport.decode({
+      headers: Object.fromEntries(request.headers.entries()),
+      body: request.body
+        ? await toBuffer(toIt(request.body))
+        : Uint8Array.from([]),
+    })
+    if (invocations.length !== 1) {
```

Comment:
We should requests to carry a single invocations, it's been designed to send batches of invocations in a single request.

Also I think you don't need to do most of these, you could simply define a server just like you usually would except don't wrap handler in [`provider`](https://github.com/web3-storage/ucanto/blob/main/packages/server/src/handler.js) that way handler will be called without performing any kind of validation and you'll be able to execute it on desired connection.

https://github.com/web3-storage/w3protocol/blob/4cfe312c9501c2c595a2c7f3fe768d58be1dbc7b/packages/access-api/src/service/index.js#L24-L31